### DISCUSSION TITLE
Upgarding supabase postgres to 15.7 ?

### DISCUSSION THREAD

**Discussion by tlecheva** [supabase/supabase#26962] [Feature Requests]
Hello,
Supabase is great, but the latest version of postgresql 15, is 15.7.
And currently if we run supabase/postgres service in docker, we can see it is based on Postgres 15.1.x
Is there a roadmap to upgrade current supabase/docker image to supabase/postgres: 15.7 ?
Ideally, in 6 months or one year maybe ?
Or is there a way to contribute in some ways by generating and testing a new supabase/postgres: 15.7 image.
Thanks & take care,
Thomas

**Comment by vickkhera** [comment]
I'm really curious why Supabase holds back on the version like this. The Postgres project is *extremely* careful of compatibility with minor version number releases. In 25+ years of self-hosting very large Postgres clusters, never once have I had any issues with just stop/upgrade/start of the DB.

The upgrade process within Supabase cloud already goes through a full dump/restore, so there *really* shouldn't be any issues with bumping the version.  Also, given that Postgres 17 is now in beta, that would be a better target to aim for than just 15.7. We could skip right over 16.

**Comment by acjay** [comment]
> Also, given that Postgres 17 is now in beta, that would be a better target to aim for than just 15.7. We could skip right over 16.

People are on old versions for all sorts of reasons. For a package like this, it makes more sense to try to offer an image for each Postgres release.

If this isn't something Supabase wants to be fully supporting, it might be on someone else to fork with a different perspective on what images should be available.

Or, folks can fork to release individual versions, as needed.

**Reply by AntonOfTheWoods** [reply]
I'm not aware of any good reasons not to upgrade to the latest version of 15... Did you find some public examples with Postgres where they borked with minor upgrades? In any case, the problem with this is that it fundamentally looks at supabase as a SAAS product, not an open source project. They maybe *should* be sticking to a conservative version on their SAAS product but I disagree that they should be ignoring the fact that many in the community would like to self-host updated versions. Saying "fork and maintain everything yourself" is... not particularly open source-friendly maybe? I don't ever remember seeing an actively maintained "this is our update policy" or even anything from a Supabase employee on the matter in the last couple of years, though there have definitely been tickets/branches/PRs dealing with compatibility with 16+.

That said, there is a `supabase/postgres` `15.8` image now...
